File: 812-rotate-string/rotate-string.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        len_s = len(s)
        len_goal = len(goal)

        if len_s != len_goal:
            return False

        for i in range(len_s):
            s = s[1:] + s[0]
            if s == goal:
                return True

        return False 

File: 812-rotate-string/rotate-string.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
        // for(int i = 0; i < s.size(); i++) {
        //     if(s.substr(i) + s.substr(0, i) == goal) return true;
        // }
        // return false;
        if(s.size() != goal.size()) return false;
        
        string ss = s+s;
        return ss.find(goal) != string::npos;
    }
};